src/utilities/nep_utilities.cuh

```diff
@@ -776,3 +776,76 @@ find_q_with_5body(const int n_max_angular_plus_1, const int n, const float* s, f
   q[5 * n_max_angular_plus_1 + n] = C5B[0] * s0_sq * s0_sq + C5B[1] * s0_sq * s1_sq_plus_s2_sq +
                                     C5B[2] * s1_sq_plus_s2_sq * s1_sq_plus_s2_sq;
 }
+
+#ifdef USE_TABLE
+namespace
+{
+const int table_length = 2001;
+const int table_segments = table_length - 1;
+const float table_resolution = 0.0005f;
+
+__device__ void find_index_and_weight(
+  const float d12_reduced,
+  int& index_left,
+  int& index_right,
+  float& weight_left,
+  float& weight_right)
+{
+  float d12_index = d12_reduced * table_segments;
+  index_left = int(d12_index);
+  if (index_left == table_segments) {
+    --index_left;
+  }
+  index_right = index_left + 1;
+  weight_right = d12_index - index_left;
+  weight_left = 1.0f - weight_right;
```

Comment:
From this I understand that `weight_right` is indeed the "fraction" of the right table value, and equivalently for the left. Neat!

Reply:
Rright!